src/worldcereal/job.py

```diff
@@ -218,3 +205,120 @@ def collect_inputs(
+            {"dimension": "y", "unit": "px", "value": 100},
+        ],
+        overlap=[
+            {"dimension": "x", "unit": "px", "value": 0},
+            {"dimension": "y", "unit": "px", "value": 0},
+        ],
+    )
+
+    # Run model inference on features
+    classes = apply_model_inference(
+        model_inference_class=PRODUCT_SETTINGS[WorldCerealProduct.CROPLAND][
+            "classification"
+        ]["classifier"],
+        cube=features,
+        parameters=PRODUCT_SETTINGS[WorldCerealProduct.CROPLAND]["classification"][
+            "parameters"
+        ],
+        size=[
+            {"dimension": "x", "unit": "px", "value": 100},
+            {"dimension": "y", "unit": "px", "value": 100},
+            {"dimension": "t", "value": "P1D"},
+        ],
+        overlap=[
+            {"dimension": "x", "unit": "px", "value": 0},
+            {"dimension": "y", "unit": "px", "value": 0},
+        ],
+    )
+
+    # Cast to uint8
+    classes = compress_uint8(classes)
+
+    return classes
+
+
+def _croptype_map(inputs: DataCube, cropland_mask: DataCube = None) -> DataCube:
+    """Method to produce croptype map from preprocessed inputs, using
+    a Presto feature extractor and a CatBoost classifier.
+
+    Args:
+        inputs (DataCube): preprocessed input cube
+        cropland_mask (DataCube): optional cropland mask
+
+    Returns:
+        DataCube: croptype labels and probability
+    """
+
+    # Run feature computer
+    features = apply_feature_extractor(
+        feature_extractor_class=PRODUCT_SETTINGS[WorldCerealProduct.CROPTYPE][
+            "features"
+        ]["extractor"],
+        cube=inputs,
+        parameters=PRODUCT_SETTINGS[WorldCerealProduct.CROPTYPE]["features"][
+            "parameters"
+        ],
+        size=[
+            {"dimension": "x", "unit": "px", "value": 100},
+            {"dimension": "y", "unit": "px", "value": 100},
+        ],
+        overlap=[
+            {"dimension": "x", "unit": "px", "value": 0},
+            {"dimension": "y", "unit": "px", "value": 0},
+        ],
+    )
+
+    # Run model inference on features
+    classes = apply_model_inference(
+        model_inference_class=PRODUCT_SETTINGS[WorldCerealProduct.CROPTYPE][
+            "classification"
+        ]["classifier"],
+        cube=features,
+        parameters=PRODUCT_SETTINGS[WorldCerealProduct.CROPTYPE]["classification"][
+            "parameters"
+        ],
+        size=[
+            {"dimension": "x", "unit": "px", "value": 100},
+            {"dimension": "y", "unit": "px", "value": 100},
+            {"dimension": "t", "value": "P1D"},
+        ],
+        overlap=[
+            {"dimension": "x", "unit": "px", "value": 0},
+            {"dimension": "y", "unit": "px", "value": 0},
+        ],
+    )
+
+    # Mask cropland
+    if cropland_mask is not None:
+        classes = classes.mask(cropland_mask == 0, replacement=0)
+
+    # Cast to uint16
+    classes = compress_uint16(classes)
```

Comment:
This is something that needs to be changed to uint32 right?

@jdries Is there a data type change trigger if I do, let's say
`cube = cube.linear_scale_range(0, 4_294_967_295 - 1, 0, 4_294_967_295 - 1)` ?

Reply:
(`4_294_967_295` is the uint32 max value)

Reply:
at the moment we're still fine as we just have few classes and simple integers representing them but I agree this needs to be resolved when we move to other legend.